include/seqan3/alignment/aligned_sequence/aligned_sequence_concept.hpp

```diff
@@ -272,4 +285,92 @@ inline typename seq_type::iterator erase_gap(seq_type & seq, typename seq_type::
 }
 //!\}
 
+namespace detail
+{
+
+/*!\brief               Create the formatted alignment output and add it to the provided debug_stream.
+ * \ingroup             aligned_sequence
+ * \tparam alignment_t  The type of the alignment, must satisfy tuple_like_concept.
+ * \tparam idx          An index sequence.
+ * \param[in] stream    The output stream that receives the formatted alignment.
+ * \param[in] align     The alignment that shall be streamed.
+ */
+template<tuple_like_concept alignment_t, size_t ...idx>
+void stream_alignment(debug_stream_type & stream, alignment_t const & align, std::index_sequence<idx...> const & /**/)
+{
+    size_t const alignment_size = std::get<0>(align).size();
+
+    // split alignment into blocks of length 50 and loop over parts
+    for (size_t begin_pos = 0; begin_pos < alignment_size; begin_pos += 50)
+    {
+        size_t const end_pos = std::min(begin_pos + 50, alignment_size);
+
+        // write header line
+        if (begin_pos != 0)
+            stream << '\n';
+
+        stream << std::setw(7) << begin_pos << ' ';
```

Comment:
Could we not make the indentation dependent on the value of `alignment_size`?

Reply:
Not sure how 'diffable' two outputs are, because we encode the position in the output. This is not a blocker, but I think we should open a follow-up issue to think about it.

Reply:
I don't see a benefit, as we fix each sequence line to length 50 anyway. 
But fine, we can discuss this in a separate issue #614.